src/types/DashboardVisualisation.ts

```diff
@@ -26,7 +26,9 @@ export enum DashboardItemSize {
 }
 
 export type DashboardVisualisation = {
+    id?: string;
     dataSourceId: DataSourceID;
     visualisationType: VisualisationType;
     options: ChartCustomisationOption;
+    metadataId: string;
```

Comment:
What is the purpose of this ID? Can't this be inferred from dataSourceId since each metadata is one-to-one with data sources?

Reply:
Changing to use `dataSourceId`

Reply:
Ah nevermind. The reason I added this is because when you click `edit` on a visualization from the Dashboard, you land in something like `.../explore/edit/{uuid for the given visualization}`. If you then delete the visualization, you are redirected to `.../explore/edit/{metadata id}`, since the UUID for the given visualization does not exist any longer.

Reply:
The parent of the add to dashboard button has the metadata entry, so you can pass that down to the button, and thus _should_ be able to avoid the field in the DashboardVisualisation